test/tools/constopt.h

```diff
@@ -0,0 +1,64 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include <libdevcore/CommonIO.h>
+#include <libevmasm/Assembly.h>
+#include <libsolc/libsolc.h>
+
+#include <libevmasm/ConstantOptimiser.h>
+
+#include <string>
+#include <sstream>
+#include <iostream>
+
+using namespace std;
+using namespace dev;
+using namespace dev::eth;
+
+namespace {
+    void testConstantOptimizer(string const &input, bool quiet) {
```

Comment:
I think this should just go into fuzzer.cpp

Reply:
iiuc

- current fuzzer.h/cpp become `fuzzer_common.cpp` and `fuzzer.h`
- The `main` function in current `fuzzer.cpp` -> solc_afl_fuzzer.cpp
- Include `fuzzer_common.h` and link against `fuzzer_common.cpp`
  - This applies for `solc_afl_fuzzer.cpp` as well as the other cpp files currently in the `ossfuzz` directory

Is this an accurate summary @chriseth ?

Reply:
Sounds good, alhough perhaps just `afl_fuzzer.cpp` would be a better name (or `fuzzer_afl.cpp`), because it concerns more than just solc.

Reply:
ack. What is the preferred naming convention for the ossfuzz-specific test harnesses?

Current consensus seems to be
- solc_opt_ossfuzz: Fuzzing the `solidity_compile` API with optimization on
- solc_noopt_ossfuzz: Likewise with optimization off
- const_opt_ossfuzz: Fuzzing the `ConstantOptimisationMethod::optimiseConstants` API

Do you have other thoughts?